### DISCUSSION TITLE
Using Knex with Planetscale

### DISCUSSION THREAD

**Discussion by boriskogan81** [planetscale/discussion#603] [Q&A]
Hi, guys, I'm deploying a project I've been working on for a while and have had Planetscale recommended. I'm excited to start working with it, but my project uses Knex/Bookshelf, and native Knex can't connect to Planetscale. I found this repo: https://github.com/hfaulds/knex-planetscale , but unfortunately it uses the import syntax, while my project uses require().  Is there any convenient workaround?

**Comment by orware** [comment]
Hello @boriskogan81,

It may be useful to write in to us over at https://support.planetscale.com/ if closer coordination is required with us.

However, reading through the repository link you shared, that one is quite dated at this point and the part where it mentions "it will only exist until planetscale exposes datbase URLs themselves" should be resolved now as we do provide those these days and you shouldn't need to use that separate project which should eliminate the syntax issue you were describing.

For our Node.js examples we typically provide a `DATABASE_URL` in the following format:

```
DATABASE_URL='mysql://<USERNAME>:<PASSWORD>@aws.connect.psdb.cloud/<DATABASE_NAME>?ssl={"rejectUnauthorized":true}'
```

And if Knex uses `mysql2` under the hood the above format will likely be accepted by it as well.

In their documentation, they do show within their Postgres example the use of an `ssl` option that is similar to what we provide above in the `DATABASE_URL` :
https://knexjs.org/guide/#configuration-options

```
ssl: config["DB_SSL"] ? { rejectUnauthorized: false } : false,
```

Example:
<img width="702" alt="image" src="https://github.com/planetscale/discussion/assets/931650/1982d678-73ad-449d-b5d2-4afbda2f1e33">

While their documentation doesn't show this with a MySQL example, I would expect that it should be supported for MySQL connections too if you are using passing in your variables similar to the Postgres example above.

Hopefully these tips above are helpful!

**Comment by boriskogan81** [comment]
Thank you.

I got it working with this config:

"production": {
  "client": "mysql",
  "connection": {
    "host": {{HOST}},
    "user": {{USER}},
    "password": {{PASSWORD}},
    "database": {{DATABASE}},
    "ssl": {
      "rejectUnauthorized": false
    }
  },
  "debug": true
},

Sincerely,

Baruch

**Reply by orware** [reply]
Hello @boriskogan81,

Does it also work if `rejectUnauthorized` is set to `true`?

Ideally you would want to have enabled as it is more secure since it verifies the certificate being used helping to prevent man-in-the-middle attacks.

**Comment by boriskogan81** [comment]
Yes, works. Thank you.
Sincerely,

Baruch
